projects/igniteui-angular/src/lib/calendar/navigation.service.ts

```diff
@@ -180,3 +189,39 @@ export class IgxDaysViewNavigationService {
         return false;
     }
 }
+
+/**
+ * @hidden
+ * Temporary service managing the multi view calendar interactions.
+ * https://github.com/IgniteUI/igniteui-angular/issues/6375
+ * TODO: should be combined with @IgxDaysViewNavigationService
+ */
+@Injectable()
+export class IgxCalendarNavigationService {
```

Comment:
Is it needed for this logic to be part of the Navigation service ? It brings dependency to the Calendar, and also it is just a calculation logic that has nothing to do with navigation.